src/vivarium/framework/values.py

```diff
@@ -588,10 +671,10 @@ def register_rate_producer(
     def register_value_modifier(
         self,
         value_name: str,
-        modifier: Callable,
-        requires_columns: List[str] = (),
-        requires_values: List[str] = (),
-        requires_streams: List[str] = (),
+        modifier: ValueSource | ValueMutator,
```

Comment:
I guess this is the fundamental place to consider if `ValueSource` is appropriate